components/fxa-client/src/internal/error.rs

```diff
@@ -138,12 +138,37 @@ error_support::define_error! {
         (UnexpectedStatus, viaduct::UnexpectedStatus),
         (MalformedUrl, url::ParseError),
         (SyncError, sync15::Error),
-        (ProtobufDecodeError, prost::DecodeError),
     }
 }
 
 error_support::define_error_conversions! {
     ErrorKind {
         (HawkError, hawk::Error),
+        (IntegerConversionError, std::num::TryFromIntError),
+    }
+}
+
+// The public FFI puts the errors into three buckets, this helps us
+// convert between them. Maybe in future we can use uniffi to expose
+// more error info to the caller?
+impl From<super::Error> for crate::FxaError {
```

Comment:
I good future followup would be to lean on UniFFI to send much more fine-grained errors across the FFI.